src/SOS/Strike/hostcoreclr.cpp

```diff
@@ -294,22 +300,41 @@ static HRESULT GetHostRuntime(std::string& coreClrPath, std::string& hostRuntime
     if (g_hostRuntimeDirectory == nullptr)
     {
 #ifdef FEATURE_PAL
-#if defined(__APPLE__)
-        hostRuntimeDirectory.assign("/usr/local/share/dotnet/shared/Microsoft.NETCore.App");
-#elif defined (__FreeBSD__) || defined(__NetBSD__)
-        ExtErr("FreeBSD or NetBSD not supported\n");
-        return E_FAIL;
-#else
-        // Start with the possible RHEL's locations, then the regular Linux path
-        for (int i = 0; i < _countof(g_linuxPaths); i++)
+        char* line = nullptr;
+        size_t lineLen = 0;
+
+        // Start with Linux location file if exists
+        FILE* locationFile = fopen("/etc/dotnet/install_location", "r");
+        if (locationFile != nullptr)
         {
-            hostRuntimeDirectory.assign(g_linuxPaths[i]);
-            if (access(hostRuntimeDirectory.c_str(), F_OK) == 0)
+            if (getline(&line, &lineLen, locationFile) != -1)
             {
-                break;
+                hostRuntimeDirectory.assign(line);
+                size_t newLinePostion = hostRuntimeDirectory.rfind('\n');
+                if (newLinePostion != std::string::npos) {
+                    hostRuntimeDirectory.erase(newLinePostion);
+                    hostRuntimeDirectory.append("/shared/Microsoft.NETCore.App");
+                }
+                free(line);
             }
         }
+        if (hostRuntimeDirectory.empty())
+        {
+#if defined (__FreeBSD__) || defined(__NetBSD__)
+            ExtErr("FreeBSD or NetBSD not supported\n");
```

Comment:
The new code suggests that FreeBSD or NetBSD would be supported if they used /etc/dotnet/install_location. If we don't expect these platforms to ever work I'd move the error check back to the top of the method and if we do expect them to have it I'd change the error text to say that.